bug_actiongroup.php

```diff
@@ -206,19 +208,27 @@
 		case 'UP_STATUS':
 			$f_status = gpc_get_int( 'status' );
 			if( access_has_bug_level( access_get_status_threshold( $f_status, $t_bug->project_id ), $t_bug_id ) ) {
-				if( true == bug_check_workflow( $t_status, $f_status ) ) {
+				if( true == bug_check_workflow( $t_status, $f_status ) ||
+					access_compare_level( current_user_get_access_level(), config_get_access( 'status_enum_workflow' ) ) ) {
 					# @todo we need to issue a helper_call_custom_function( 'issue_update_validate', array( $t_bug_id, $t_bug_data, $f_bugnote_text ) );
-					bug_set_field( $t_bug_id, 'status', $f_status );
-
-					# Add bugnote if supplied
-					if( !is_blank( $f_bug_notetext ) ) {
-						bugnote_add( $t_bug_id, $f_bug_notetext, null, $f_bug_noteprivate );
-						# No need to call email_generic(), bugnote_add() does it
+					if( $t_status !== $f_status ) {
+						bug_set_field( $t_bug_id, 'status', $f_status );
+						if( is_blank( $f_bug_notetext ) ) {
+							$t_new_status_label = MantisEnum::getLabel( config_get( 'status_enum_string' ), $f_status );
+							$t_new_status_label = str_replace( ' ', '_', $t_new_status_label );
+							email_bug_status_changed( $t_bug_id, $t_new_status_label );
+						} else {
+							bugnote_add( $t_bug_id, $f_bug_notetext, null, $f_bug_noteprivate );
+							# No need to call email_generic(), bugnote_add() does it
+						}
+						helper_call_custom_function( 'issue_update_notify', array( $t_bug_id ) );
```

Comment:
I'm not really sure I have enough knowledge to add this call. Is this call supposed to include previous and new bugdata?

event_signal( 'EVENT_BUG_ACTION',... called right after switch...case.